src/services/completions.ts

```diff
@@ -2464,6 +2466,29 @@ function getCompletionEntryCodeActionsAndSourceDisplay(
         return { codeActions: [codeAction], sourceDisplay: undefined };
     }
 
+    if (contextToken && isObjectLiteralExpression(contextToken.parent.parent) && previousToken?.kind !== SyntaxKind.ColonToken) {
+        let changes: FileTextChanges[];
+        if (getLineAndCharacterOfPosition(sourceFile, contextToken.getEnd()).line !== getLineAndCharacterOfPosition(sourceFile, position).line) {
+            changes = textChanges.ChangeTracker.with(
+                { host, formatContext, preferences },
+                tracker=>tracker.replacePropertyAssignment(sourceFile, contextToken.parent as PropertyAssignment ,contextToken.parent as PropertyAssignment));
+        }
+        else {
+            changes = textChanges.ChangeTracker.with(
+                { host, formatContext, preferences },
+                tracker=>tracker.replacePropertyAssignmentOnSameLine(sourceFile, contextToken.parent as PropertyAssignment ,contextToken.parent as PropertyAssignment));
+        }

```

Comment:
This, and the addition of `replacePropertyAssignmentOnSameLine`, isn’t worth the value you’re getting out of `replacePropertyAssignment`. You’re not even really replacing the property assignment, since `oldNode` and `newNode` are identical in both of these calls. Once you move the check for whether a comma insertion is needed to `createCompletionEntry` as I suggested in my previous comment, it will make even less sense to use these functions, since currently the check for an existing comma is tangled up with the code to actually insert it. Instead, to check for a comma, just do something like

```ts
findNextToken(propertyAssignment)?.kind === SyntaxKind.CommaToken
```

and to insert a comma, just use something like `changes.insertText(sourceFile, propertyAssignment.end, ",")`.